diff --git a/xsys_linux.c b/xsys_linux.c
index 51ea073..82823ab 100644
--- a/xsys_linux.c
+++ b/xsys_linux.c
@@ -29,7 +29,9 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
+#include <termios.h>
 #include <sys/file.h>
+#include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -37,6 +39,7 @@
 int xsys_serialSetup(struct xbee_serialInfo *info) {
 	struct termios tc;
 	speed_t chosenbaud;
+	int flags;
 	
 	if (!info) return XBEE_EMISSINGPARAM;
 	
@@ -136,11 +139,7 @@ int xsys_serialSetup(struct xbee_serialInfo *info) {
 	tc.c_cflag &= ~ CSIZE;            /* remove size flag... */
 	tc.c_cflag |=   CS8;              /* ...enable 8 bit characters */
 	tc.c_cflag |=   HUPCL;            /* enable lower control lines on close - hang up */
-#ifdef XBEE_NO_RTSCTS
 	tc.c_cflag &= ~ CRTSCTS;          /* disable hardware CTS/RTS flow control */
-#else
-	tc.c_cflag |=   CRTSCTS;          /* enable hardware CTS/RTS flow control */
-#endif
 	/* local flags */
 	tc.c_lflag &= ~ ISIG;             /* disable generating signals */
 	tc.c_lflag &= ~ ICANON;           /* disable canonical mode - line by line */
@@ -162,6 +161,11 @@ int xsys_serialSetup(struct xbee_serialInfo *info) {
 		perror("tcsetattr()");
 		return XBEE_ESETUP;
 	}
+
+	/* clear RTS */
+	ioctl(info->dev.fd, TIOCMGET, &flags);
+	flags &= ~TIOCM_RTS;
+	ioctl(info->dev.fd, TIOCMSET, &flags);
 	
 	/* enable input & output transmission */
 #ifdef linux
